rfcs/0003-currency.md

```diff
@@ -0,0 +1,161 @@
+- `Native Currency`. Currency which is used to pay for operation on platforms. There could be several layers of native currency in single consensus. Native currency of consensus and native currency of protocol built on top.
+
+Above properies apply to local currencies, so to make econimics operate, need to integrate external sources of currencies to. Here we obtain more propertis.
+
+- `Remote Currency Id`. We can have local currency id. Using some forms of cross consesnsus mechanism (like relayers) we can `transfer` currency from one chain to other. So currency may have `Chain Id` from which it came along with id it has on that chain. 
+- `Remote Currency Decimals`. When transfer happens, we should know what  given `Amount` transfered means for that currency on our local netowrk. We need to know what are inimal amounts and decimals currency has remotely. This works off chain sources too, like Oracles. This may lead to precision loss and loss of currency too. In worth case transfer back and forth will print money. Example, remotely BTC can have 8 decimals, while locally we can have 12.
+- `Trusted Currency`. If there is surce of currency, like Ethereum. And there is trusted consensus (parachain) which can transfer `Etherium` currency into Polkadot relay chains ecosystem. Than that trusted consesus can be used to transfer remote currency to local with 1 to 1 correspondance. So lack of reverse of trust is open question. Process can be improved by setting up oracles which monitor burn proves on each chain. In theory burns and proves could be manual to allow manual.
+- `Mapped currency`.  Given possible lack of trust and difference of decimals, we come up the need for market to decide how well these are aligned. So if remote yBTC has local id of 42 and remote id of (200, 13) with 8 decimals and local xBTC has id of 1 and 12 decimals, than local DEX pool ratio of 10_000 should be created. Large enough pools are quantive trust. Automatic mapping and User Interface can help to handle.
+- `Remote Symbols`. 
+
+All currencies are derived, so some currencies can be derived in local consensus. So:
+- `Dillution`. If currency was direcly derived from protocol in local consensus than can tell exact ration of one amount needed to swap for other amount. With time currency amounts can change (locked, unlocked, minted), which will changed dillution factor. So we cannot tell nothing about external currencies dillution factors, only Oracles can tell. In this case we can observer whole chain of of tokens as they wrap each other and dillute. Process is not nessecary bidirectional.
+- `Local Simple Dillution` - when for each amount there is mint of other amount, may be allowing different reserver/hold propreties. May eventually be not one to one as other can be burnt. But because of can witness total supply, can say ratio. So what to do if curreny transfered from and to system?
+

```

Comment:
still in progress, as i saw karel did some improvments; i did to testing currencies. acala is good, but seems we are not compatible with their approach